velox/connectors/hive/HiveDataSink.h

```diff
@@ -262,6 +267,7 @@ class HiveInsertTableHandle : public ConnectorInsertTableHandle {
   const std::shared_ptr<HiveBucketProperty> bucketProperty_;
   const std::optional<common::CompressionKind> compressionKind_;
   const std::unordered_map<std::string, std::string> serdeParameters_;
+  const std::shared_ptr<dwio::common::WriterOptions> writerOptions_;
```

Comment:
@pedroerp Hi Pedro, I just bumped into this code.

`WriterOptions` includes some contextual information (memory pool, spill config, etc.) so I guess it's the reason why it's not included in TableWriteNode's JSON serde, see [code](https://github.com/facebookincubator/velox/blob/2ede38aad78a56103091322a1db9bdefd7252a2c/velox/connectors/hive/HiveDataSink.cpp#L1061).

Do you think this is an issue? Given it's a special case that we include contextual information in query plan node.

Reply:
Filed an issue: https://github.com/facebookincubator/velox/issues/12437